pkg/pulumiyaml/run.go

```diff
@@ -2022,27 +2037,30 @@ func (e *programEvaluator) evaluateBuiltinInvoke(t *ast.InvokeExpr) (interface{}
 		}
 
 		pkgRef := ""
-		secret, err := e.pulumiCtx.InvokePackageRaw(string(functionName), args[0], &result, pkgRef, opts...)
+		secret, deps, err := e.pulumiCtx.InvokePackageRawWithDeps(string(functionName), args[0], &result, pkgRef, opts...)
 		if err != nil {
 			return e.error(t, err.Error())
 		}
 
 		if t.Return.GetValue() == "" {
+			output := pulumi.OutputWithDependencies(context.TODO(), pulumi.Any(result), deps...)
```

Comment:
`e.pulumiCtx` has a `.Context()` which might be a good thing to hang this off, rather than `TODO`?

Reply:
Yep, there's another call to context.TODO just above in the call to `ResolveFunction`, I'll update that one too.